prow/config.yaml

```diff
@@ -74,37 +75,19 @@ branch-protection:
             <<: *blocked_branches
             master:
               protect: true
-        istio.io:
-          protect: false
-          branches:
-            <<: *blocked_branches
```

Comment:
I missed/forgot to note this in the other PR: the branch protector requires branches to be listed explicitly, `release-*` will not work, you need an entry for `release-1.0` etc. 

I don't see anything in the https://github.com/kubernetes/test-infra/tree/master/prow/cmd/branchprotector git history to suggest someone added this support